pyflakes/test/test_api.py

```diff
@@ -255,6 +256,20 @@ def test_syntaxErrorNoOffset(self):
              "bad line of source\n"),
             err.getvalue())
 
+    def test_syntaxErrorNoText(self):
+        """
+        C{syntaxError} doesn't include text or nonsensical offsets if C{text} is C{None}.
+
+        This typically happens when checking input from stdin.
+        """
+        err = io.StringIO()
+        reporter = Reporter(None, err)
+        reporter.syntaxError('<stdin>', 'a problem', 0, 0,
+                             None)
```

Comment:
this should generate an actual syntax error and validate it so we can know when to delete this behaviour (3.9+)

Reply:
This is testing the reporting of a syntax error. Specifically, the behavior of what to do when the offsets are nonsensical (which they are for actual decoding errors in all versions of python including 3.9+). I'll attempt to clarify this in the test comment.

Reply:
ah I see -- also I think the `None` fits on the line above within the character limit too if you want to move that as well!  ideally we should set up some auto-formatting